packages/fast-colors/src/colorlib/colorLAB.ts

```diff
@@ -0,0 +1,45 @@
+// tslint:disable:member-ordering
+import { roundToPrecisionSmall } from "./mathUtils";
+
+// CIELAB color space
+// https://en.wikipedia.org/wiki/CIELAB_color_space
+// This implementation uses the D65 constants for 2 degrees. That determines the constants used for the pure white point of the XYZ space of 0.95047, 1.0, 1.08883.
+// https://en.wikipedia.org/wiki/Illuminant_D65
+// These constants determine how the XYZ, LCH and LAB colors convert to/from RGB.
+export class ColorLAB {
+    public static readonly epsilon: number = 216 / 24389;
+    public static readonly kappa: number = 24389 / 27;
+
+    public static fromObject(data: any): ColorLAB | null {
+        if (data && !isNaN(data.l) && !isNaN(data.a) && !isNaN(data.b)) {
+            return new ColorLAB(data.l, data.a, data.b);
+        }
+        return null;
+    }
+
+    constructor(l: number, a: number, b: number) {
+        this.l = l;
+        this.a = a;
+        this.b = b;
+    }
+
+    public readonly l: number;
+    public readonly a: number;
+    public readonly b: number;
+
+    public equalValue(rhs: ColorLAB): boolean {
+        return this.l === rhs.l && this.a === rhs.a && this.b === rhs.b;
+    }
+
+    public roundToPrecision(precision: number): ColorLAB {

```

Comment:
roundToPrecisionSmall does the same thing as Math.Round(int precision) in C#. That is, it gives you "precision" number of digits to the right of the decimal point (which differs from javascript toFixed which returns a string with the specified number of digits combined across both left and right of the decimal point). The reason the function is named roundToPrecisionSmall is because the function will fail for very large numbers (ones where i*10^(precision+1) is larger than the max value of number). But in the color types we know the numbers will not be anywhere near that max.

Reply:
Got it - okay cool, thanks.